packages/protocol/test/signal/SignalService.integration.test.ts

```diff
@@ -0,0 +1,242 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { TestHeaderSync } from "../../typechain";
+import { deploySignalService, getSignalProof } from "../utils/signal";
+import deployAddressManager from "../utils/addressManager";
+import { getBlockHeader } from "../utils/rpc";
+import {
+    getDefaultL2Signer,
+    getL1Provider,
+    getL2Provider,
+} from "../utils/provider";
+
+describe("integration:SignalService", function () {
+    async function deployIntegrationSignalService() {
+        const [owner, nonOwner] = await ethers.getSigners();
+
+        const { chainId: srcChainId } = await ethers.provider.getNetwork();
+
+        // ethereum node
+        const l1Provider = await getL1Provider();
+
+        // hardhat node
+        const l2Provider = await getL2Provider();
+
+        const l1Signer = await ethers.provider.getSigner();
+
+        const l2Signer = await getDefaultL2Signer();
+
+        const l2NonOwner = await l2Provider.getSigner();
+
+        const l2Network = await l2Provider.getNetwork();
+        const enabledDestChainId = l2Network.chainId;
+
+        const addressManager = await deployAddressManager(l1Signer);
+        const l2AddressManager = await deployAddressManager(l2Signer);
+
+        const { signalService: l1SignalService } = await deploySignalService(
+            owner,
+            addressManager,
+            srcChainId
+        );
+
+        const { signalService: l2SignalService } = await deploySignalService(
+            l2Signer,
+            l2AddressManager,
+            enabledDestChainId
+        );
+
+        await addressManager.setAddress(
+            `${enabledDestChainId}.signal_service`,
+            l2SignalService.address
+        );
+
+        await l2AddressManager.setAddress(
+            `${srcChainId}.signal_service`,
+            l1SignalService.address
+        );
+
+        const headerSync: TestHeaderSync = await (
+            await ethers.getContractFactory("TestHeaderSync")
+        )
+            .connect(l2Signer)
+            .deploy();
+
+        await l2AddressManager.setAddress(
+            `${enabledDestChainId}.taiko`,
+            headerSync.address
+        );
+
+        return {
```

Comment:
If this method is only used in this files and only requires to provide these values, you can simplify the return statement by only returning those values.

```
 l1Provider,
            owner,
            l1SignalService,
            l2SignalService,
            enabledDestChainId,
            headerSync,
```